docs/source/cpp/tables.rst

```diff
@@ -77,6 +77,17 @@ has a schema which must match its arrays' datatypes.
 Record batches are a convenient unit of work for various serialization
 and computation functions, possibly incremental.
 
+.. image:: tables-versus-record-batches.svg
+   :alt: A graphical representation of an Arrow Table and a Record Batch, with
+         structure as described in text above.
+
+Because record batches can be represented as a struct array, they can be 
+exported through the C data interface between implementations. Tables and 
+chunked arrays, on the other hand, are concepts in the C++ implementation, not 
```

Comment:
Knowledge of the C data interface aside, it may be worth being explicit where Tables can and cannot go -- for example, Python claims to have a Table class which is described similarly, and this seems like a good moment to be clear about incompatibilities/compatibilities.

Reply:
Yeah maybe it should be C++ and it's bindings (Python is one of those).

Another diagram I'd like to have is which implementations are base ones (e.g. C++, Rust), and which are bindings (and to whom). I know where C++/Python/R/Ruby, C#, and Rust sit. I'm still unclear where Go and Java are though.

Reply:
Go and Java are standalone implementations like C++/C♯/Rust/Julia (Java was one of the original implementations)